crates/common/src/hashchain.rs

```diff
@@ -220,6 +230,20 @@ impl Hashchain {
             .unwrap_or(false)
     }
 
+    fn verify_signature_with_key_at(
```

Comment:
I would rename to `verify_signature_at_key_idx`, and either:
1. Make `idx` a u64 and convert to usize inline 
2. Make `SignatureBundle` use usize

Reply:
opted for 2, because we are using `Vec<HashchainEntry>` atm, which would cause problems when we were on a non-x64 platform.